File: Lista de Exercicio Paradigmas/Lista Python/Ex13.py
class Matematica:
    @staticmethod
    def fatorial(n):
        if n < 0:
            raise ValueError("O fatorial não está definido para números negativos.")
        elif n == 0 or n == 1:
            return 1
        else:
            resultado = 1
            for i in range(2, n + 1):
                resultado *= i
            return resultado

# Exemplo de uso
numero = 5
print(f"O fatorial de {numero} é {Matematica.fatorial(numero)}.")

File: Lista de Exercicio Paradigmas/Lista Java/Ex13.java
class Matematica {
    public static int fatorial(int n) {
        if (n < 0) {
            throw new IllegalArgumentException("O fatorial não está definido para números negativos.");
        } else if (n == 0 || n == 1) {
            return 1;
        } else {
            int resultado = 1;
            for (int i = 2; i <= n; i++) {
                resultado *= i;
            }
            return resultado;
        }
    }

    public static void main(String[] args) {
        int numero = 5;
        System.out.println("O fatorial de " + numero + " é " + Matematica.fatorial(numero) + ".");
    }
}
